### PR TITLE
TMA-575: Add support for raw export-report API

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

File: lib/gooddata/rest/connection.rb [hunk 1]
```diff
@@ -288,14 +291,16 @@ def download(what, where, options = {})
             :verify_ssl => verify_ssl
           }
           RestClient::Request.execute(raw) do |chunk, _x, response|
-            if response.code.to_s != '200'
+            if response.code.to_s == '202'
```
Comment (yenkins) [inline lib/gooddata/rest/connection.rb]: Calls 'response.code' 3 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/rest/connection.rb]: Calls 'response.code.to_s' 2 times (DuplicateMethodCall)

File: lib/gooddata/rest/connection.rb [hunk 2]
```diff
@@ -590,6 +595,9 @@ def process_response(options = {}, &block)
         elsif content_type == 'application/zip'
           result = response
           GoodData.rest_logger.debug 'Response: a zipped stream'
+        elsif content_type == 'text/csv'
+          result = response
+          GoodData.rest_logger.debug 'Response: CSV text'
```
Comment (yenkins) [inline lib/gooddata/rest/connection.rb]: Calls 'gooddata.rest_logger' 7 times (DuplicateMethodCall)

File: lib/gooddata/rest/connection.rb [hunk 3]
```diff
@@ -59,6 +59,9 @@ def get_redirection_cookies(request, _result, _args)
 
 module GoodData
   module Rest
+    class RestRetryError < StandardError
```
Comment (yenkins) [inline lib/gooddata/rest/connection.rb]: Has no descriptive comment (IrresponsibleModule)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/625/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/419/ : SUCCESS in 2m 07s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/727/ : SUCCESS in 1m 57s

**Comment by panjan** [comment]
extended test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/626/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/420/ : SUCCESS in 1m 52s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/728/ : SUCCESS in 1m 57s

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/627/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1151/ : FAILURE in 50m 19s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/909/ : SUCCESS in 31m 37s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/628/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/421/ : SUCCESS in 1m 55s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/729/ : SUCCESS in 32s

**Comment by panjan** [comment]
extended test

File: lib/gooddata/models/metadata/report.rb [hunk 4]
```diff
@@ -174,6 +174,16 @@ def export(format, options = {})
...
     end
 
+    # Exports a report too large to be computed on the UI. It executes in raw form.
+    # Saves the result into file.
+    #
+    # @param filename [String] Filename to save into
+    # @return [Net::HTTPResponse] Returns HTTP status
+    def export_raw(filename)
+      result = client.post("/gdc/app/projects/#{project.pid}/execute/raw", 'report_req' => { 'report' => uri })
+      client.download(result['uri'], filename, { url_encode: false })
```
Comment (yenkins) [inline lib/gooddata/models/metadata/report.rb]: Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/629/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/422/ : FAILURE in 1m 52s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/730/ : SUCCESS in 1m 47s

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/630/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1152/ : SUCCESS in 49m 55s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/910/ : SUCCESS in 32m 00s

File: lib/gooddata/models/metadata/report.rb [hunk 5]
```diff
@@ -174,6 +174,16 @@ def export(format, options = {})
...
     end
 
+    # Exports a report too large to be computed on the UI. It executes in raw form.
+    # Saves the result into file.
+    #
+    # @param filename [String] Filename to save into
+    # @return [Net::HTTPResponse] Returns HTTP status
+    def export_raw(filename)
+      result = client.post("/gdc/app/projects/#{project.pid}/execute/raw", 'report_req' => { 'report' => uri })
+      client.download(result['uri'], filename, url_encode: false )
```
Comment (yenkins) [inline lib/gooddata/models/metadata/report.rb]: Style/SpaceInsideParens: Space inside parentheses detected.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/631/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/423/ : FAILURE in 36s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/731/ : SUCCESS in 35s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/632/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/424/ : SUCCESS in 34s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/732/ : SUCCESS in 1m 46s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/241/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/425/ : SUCCESS in 2m 56s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/733/ : SUCCESS in 43s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1153/ : FAILURE in 52m 40s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/911/ : SUCCESS in 37m 25s

**Comment by panjan** [comment]
extended test - staging

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/634/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1154/ : SUCCESS in 44m 17s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/912/ : SUCCESS in 30m 20s

**Pull request merged**